reconciler/reconcile_common.go

```diff
@@ -49,7 +49,9 @@ func PostProcessReconcile(ctx context.Context, resource duckv1.KRShaped) {
 	newStatus.ObservedGeneration = resource.GetGeneration()
 
 	rc := mgr.GetTopLevelCondition()
-	if rc.Reason == failedGenerationBump {
+	if rc == nil {
```

Comment:
Fold the assignment in?
```suggestion
	if rc := mgr.GetTopLevelCondition(); rc == nil {
```